micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/UrlMetricNameProvider.groovy

```diff
@@ -0,0 +1,63 @@
+package com.ofg.infrastructure.web.resttemplate.fluent
+
+import groovy.transform.CompileStatic
+
+import java.util.regex.Pattern
+
+import static com.google.common.base.Preconditions.checkArgument
+import static org.springframework.util.StringUtils.hasText
+import static org.springframework.util.StringUtils.trimLeadingCharacter
+import static org.springframework.util.StringUtils.trimTrailingCharacter
+
+@CompileStatic
```

Comment:
As you know we're in the middle of rewriting micro-infra-spring to Java.
Could you rewrite `UrlMetricNameProvider` to Java